service/discovery/azure/storage.go

```diff
@@ -113,9 +114,107 @@
 	}
 	list = append(list, dbs...)
 
+	// Discover Cosmos DB
+	cosmosDB, err := d.discoverCosmosDB()
+	if err != nil {
+		return nil, fmt.Errorf("could not discover sql databases: %w", err)
+	}
+	list = append(list, cosmosDB...)
+
 	return
 }
 
+// discoverCosmosDB discovers Cosmos DB accounts
+func (d *azureStorageDiscovery) discoverCosmosDB() ([]voc.IsCloudResource, error) {
+	var (
+		list []voc.IsCloudResource
+		err  error
+	)
+
+	// initialize Cosmos DB client
+	if err := d.initCosmosDBClient(); err != nil {
+		return nil, err
+	}
+
+	// Discover Cosmos DB
+	err = listPager(d.azureDiscovery,
+		d.clients.cosmosDBClient.NewListPager,
+		d.clients.cosmosDBClient.NewListByResourceGroupPager,
+		func(res armcosmos.DatabaseAccountsClientListResponse) []*armcosmos.DatabaseAccountGetResults {
+			return res.Value
+		},
+		func(res armcosmos.DatabaseAccountsClientListByResourceGroupResponse) []*armcosmos.DatabaseAccountGetResults {
+			return res.Value
+		},
+		func(dbAccount *armcosmos.DatabaseAccountGetResults) error {
+			cosmos, err := d.handleCosmosDB(dbAccount)
+			if err != nil {
+				return fmt.Errorf("could not handle sql database: %w", err)
+			}
+			log.Infof("Adding Cosmos DB account '%s", *dbAccount.Name)
+			list = append(list, cosmos)
+
+			return nil
+		})
+	if err != nil {
+		return nil, err
+	}
+
+	return list, nil
+}
+
+func (d *azureStorageDiscovery) handleCosmosDB(account *armcosmos.DatabaseAccountGetResults) (voc.IsCloudResource, error) {
+	var (
+		enc voc.IsAtRestEncryption
+		err error
+	)
+
+	// initialize Cosmos DB client
+	if err = d.initCosmosDBClient(); err != nil {
+		return nil, err
+	}
+
+	// Check if KeyVaultURI is set
+	// By default the Cosmos DB account is encrypted by Azure managed keys. Optionally, it is possible to add a second encryption layer with customer key encryption. (see https://learn.microsoft.com/en-us/azure/cosmos-db/how-to-setup-customer-managed-keys?tabs=azure-portal)
+	if account.Properties.KeyVaultKeyURI != nil {
+		enc = &voc.CustomerKeyEncryption{
+			AtRestEncryption: &voc.AtRestEncryption{
+				Enabled: true,
+				// Algorithm: algorithm, //TODO(anatheka): How do we get the algorithm? Are we available to do it by the related resources?
```

Comment:
Yes, it seems so. The other option would be to do an additional redundant call to keyvault/keys client (redundant because we do it in KeyVaults already). So this should be a perfect fit for related resources.

For now it is enough to see if it is enabled I think.